### PR TITLE
support xunit output from spacejam process

### PR DESCRIPTION
**Pull request by meonkeys**
This is an alternative to PR #28.

Example xunit XML file output:

    <testsuite errors="" failures="" name="meteor" skips="" tests="" time="">
    <testcase classname="tinytest.server-only-test-server" name="server-only-test-server" time="0.001">
    </testcase>
    </testsuite>

### CODE REVIEW

File: src/Phantomjs.coffee [hunk 1]
```diff
@@ -25,14 +28,21 @@ class Phantomjs extends EventEmitter
...
     spawnArgs = options.split(' ')
     spawnArgs.push(script)
+    if xunitOutput
+      # set platform to xunit (see phantomjs-test-in-console)
+      spawnArgs.push('xunit')
     spawnOptions = {
       cwd: __dirname,
       detached: false
       env: env
+      pipeToFile: xunitOutput
```
Comment (rbabayoff) [inline src/Phantomjs.coffee]: No need for this.

File: src/Phantomjs.coffee [hunk 2]
```diff
@@ -12,11 +14,12 @@ class Phantomjs extends EventEmitter
 
   childProcess: null
 
-  run: (url, options = '--load-images=no --ssl-protocol=TLSv1', script = "phantomjs-test-in-console")=>
+  run: (url, options = '--load-images=no --ssl-protocol=TLSv1', script = "phantomjs-test-in-console", xunitOutput = '')=>
```
Comment (rbabayoff) [inline src/Phantomjs.coffee]: Phatnomjs is a generic class that shouldn't be aware of xunit output. I'd rather change this from xunitOutput to a spawnArgs array

File: src/Phantomjs.coffee [hunk 3]
```diff
@@ -25,14 +28,21 @@ class Phantomjs extends EventEmitter
...
     spawnOptions = {
       cwd: __dirname,
       detached: false
       env: env
+      pipeToFile: xunitOutput
     }
 
     @childProcess = new ChildProcess()
-    @childProcess.spawn("phantomjs", spawnArgs, spawnOptions)
+    pipeClass = Pipe
```
Comment (rbabayoff) [inline src/Phantomjs.coffee]: pipeClass should also be provided as input arg to method.

File: src/Spacejam.coffee [hunk 4]
```diff
@@ -75,7 +75,7 @@ class Spacejam extends EventEmitter
     @meteor.on "ready", =>
       log.info "spacejam: meteor is ready"
 
-      @runPhantom(@meteor.options["root-url"], options['phantomjs-options'], options['phantomjs-script'])
+      @runPhantom(@meteor.options["root-url"], options['phantomjs-options'], options['phantomjs-script'], options['xunit-output'])
```
Comment (rbabayoff) [inline src/Spacejam.coffee]: So, all xunit logic, of scriptArgs and pipeClass should happen here.

**Comment by rbabayoff** [comment]
Hey, first of, thanks for that. I made a few comments on your PR. Also, spacejam is 100% unit tested using mocha, so you'll need to write tests too. To run them, just:

```
npm test
# Or
cake test
```

Or look in Cakefile for the mocha command line we use.

**Comment by meonkeys** [comment]
Please see 941ec5dfde54ed99fd39a8799ab4d62f9519f2d7. I got to everything except tests. Am I on the right track?

I'm working on a new test locally: `XunitFilePipeTest` (`tests/src/XunitFilePipeTest.coffee`).

Looks like files in `lib/` were automatically modified. Should I commit those changes?

Also, TravisCI lied, the build should pass. GitHub is getting nailed by [a huge DDoS attack](https://github.com/blog/1981-large-scale-ddos-attack-on-github-com).

**Comment by rbabayoff** [comment]
First, yes, you are on the right track. Yes, lib files should be committed even though they're auto-generated. I didn't find a way of ignoring them in git but including them in the npm package.

**Comment by meonkeys** [comment]
I added `XunitFilePipeTest`. Works, but I couldn't find a way around using `setTimeout`. Any ideas? Any thoughts in general about 15308c4c89a5362892f2775cde6f213bbf6e17c1 and the state of the PR now?

**Comment by rbabayoff** [comment]
@meonkeys sorry for the late reply, I was on vacation. What do you mean in setTimeout? An async test?

**Comment by meonkeys** [comment]
I'm talking about [this part of XunitFilePipeTest](https://github.com/practicalmeteor/spacejam/pull/29/files#diff-0a073d4a2ad5abb1f62bb899383bb83eR20) ... it seems wrong to use `setTimeout()` there, but it breaks without it. I don't know why. If the `setTimeout()` is acceptable in the test, merge away!

**Comment by rbabayoff** [comment]
I know how to fix it. I'll do it. Give me a couple of days.

**Comment by meonkeys** [comment]
Ok, thank you.

**Comment by mcbain** [comment]
+1 :)

**Comment by rbabayoff** [comment]
Hey, I refactored the pr in #32 and also created a complete "functional" test. Published as v1.2.0. Sorry it took so long.

**Comment by meonkeys** [comment]
:cool:

**Pull request merged**